### ISSUE TITLE
Problem to run pg_activity when there's a barman connection

### ISSUE THREAD

**Issue by DenisMedeirosBBD** [dalibo/pg_activity#203]
We have a CentOS 7 server running PostgreSQL version 12 and we use Barman (https://www.pgbarman.org/) as one of our backup tools.

When we try to run pg_activity without any additional parameter, we get the following error:

```bash
[root@server ~]# sudo -u postgres pg_activity -U postgres
Traceback (most recent call last):
  File "/bin/pg_activity", line 11, in <module>
    load_entry_point('pg-activity==2.1.3', 'console_scripts', 'pg_activity')()
  File "/usr/lib/python3.6/site-packages/pgactivity/cli.py", line 293, in main
    ui.main(term, dataobj, host, options, dsn)
  File "/usr/lib/python3.6/site-packages/pgactivity/ui.py", line 227, in main
    width=width,
  File "/usr/lib/python3.6/site-packages/pgactivity/views.py", line 526, in screen
    width=width,
  File "/usr/lib/python3.6/site-packages/pgactivity/views.py", line 123, in wrapper
    for line in func(term, *args, **kwargs):
  File "/usr/lib/python3.6/site-packages/pgactivity/views.py", line 378, in processes_rows
    cell(getattr(process, field), column)
  File "/usr/lib/python3.6/site-packages/pgactivity/views.py", line 363, in cell
    text_append(f"{color}{column.render(value)}")
  File "/usr/lib/python3.6/site-packages/pgactivity/types.py", line 239, in render
    return self.template_h % self.transform(value)[: self.max_width]
  File "/usr/lib/python3.6/site-packages/pgactivity/utils.py", line 79, in ellipsis
    lv = len(v)
TypeError: object of type 'NoneType' has no len()
```

To figure out which database name was coming as `None` and making the function `ellipsis` to fail, I edited the file `types.py` to not apply the function `ellipsis` on the database name. Then, pg_activity ran well and showed a barman connection but with the database name being `None` in fact.

![image](https://user-images.githubusercontent.com/46945090/112565874-3430e900-8d9b-11eb-9556-100dc6728da2.png)

At this point I don't know where the problem is happening and why that "zombie connection" (with no reads and no writes) shows up without a database name. One possible solution would be to handle the situations when the database name is None. Something like that:

```python
def ellipsis(v: str, width: int) -> str:
    """Shorten a string to specified width with '...' in the middle.
    >>> ellipsis("loooooooooog", 7)
    'lo...og'
    >>> ellipsis("loooooooooog", 6)
    'lo...g'
    >>> ellipsis("short", 10)
    'short'
    """
    if v is None:
        return "Something here"
    lv = len(v)
    if lv <= width:
        return v
    assert width >= 5
    wl = (width - 3) // 2
    return v[: wl + 1] + "..." + v[-wl:]
```

Thanks in advance for any thought or help in how to fix this issue.

**Comment by dlax** [comment]
So it seems that the `datname` field from `pg_stat_activity` may be `NULL` in some cases (streaming replication?)?

How did you configure barman? Especially, the `conninfo` value in `pg` section described at http://docs.pgbarman.org/release/2.12/#preliminary-steps.

**Comment by tgdnts** [comment]
The "zombie" conneciton @DenisMedeirosBBD mentioned is how Postgres does streaming replication (more up-to-date than file-based logs).
Per Postgres documentation, "replication connections do not specify any particular database", which is why there's no database specified in Barman's streaming_conninfo setting.

**Comment by blogh** [comment]
Right now, I am not sure theses lines should be displayed because they are not "client backends" but "walsenders".
Given that we are displaying queries only client backend en parallel workers make sense.

The line is visible because "state <> 'idle'" filter all processes with a NULL state ... but the walsender who has a status.
It's not completely wrong since walsenders count towards max_connections but I dont like the fact that it's mixed with "normal connexions". Added "bonus" we can kill it with the software.

I think it's a good idea to Fix the null datname nonetheless because I would love to remove this limitation and display all processes in the future since the are gradually added to pg_stat_activity and we have some nice info available for them.

**Comment by dlax** [comment]
Just released version 2.1.4 including a fix for this issue.

**Comment by tgdnts** [comment]
Thank you!
